File: packages/js-dpp/lib/dataTrigger/DataTriggerExecutionResult.js
class DataTriggerExecutionResult {
  /**
   * @param {AbstractDataTriggerError[]} errors
   */
  constructor(errors = []) {
    this.errors = errors;
  }

  /**
   * Add an error to result
   *
   * @param {AbstractDataTriggerError} error
   */
  addError(...error) {
    this.errors.push(...error);
  }

  /**
   * Get all data trigger execution errors
   *
   * @return {AbstractDataTriggerError[]}
   */
  getErrors() {
    return this.errors;
  }

  /**
   * Check if result have no errors
   *
   * @return {boolean}
   */
  isOk() {
    return this.errors.length === 0;
  }
}

module.exports = DataTriggerExecutionResult;

File: packages/rs-dpp/src/data_trigger/data_trigger_execution_result.rs
use crate::errors::StateError;

#[derive(Debug, Default)]
pub struct DataTriggerExecutionResult {
    pub errors: Vec<StateError>,
}

impl From<Vec<StateError>> for DataTriggerExecutionResult {
    fn from(errors: Vec<StateError>) -> Self {
        DataTriggerExecutionResult { errors }
    }
}

impl DataTriggerExecutionResult {
    /// Add an error to result
    pub fn add_error(&mut self, err: StateError) {
        self.errors.push(err)
    }

    /// Get all Trigger Execution Errors
    pub fn get_errors(&self) -> &Vec<StateError> {
        &self.errors
    }

    /// Check if result has no errors
    pub fn is_ok(&self) -> bool {
        self.errors.is_empty()
    }
}
